### PR TITLE
MAINT: Replace np.max with np.maximum

### LINKED ISSUE #12696: scipy.optimize.fminbound, numpy depreciation warning Creating an ndarray from ragged nested sequences
When using the function `scipy.optimize.fminbound`, I get the following warning:
```
Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
```

I tracked down the message to this line:
https://github.com/scipy/scipy/blob/28d4244b6f5c1543755e586f27f9261d0f84153b/scipy/optimize/optimize.py#L1976

It seems that `np.maximum` should be used instead of `np.max` here anyway !

A minimal example:
```
from scipy import optimize
import numpy as np
optimize.fminbound(lambda x: np.array([x**2]), -np.pi, np.pi, disp=False)
```

Versions:
`scipy: 1.5.0   numpy: 1.19.1   sys: sys.version_info(major=3, minor=6, micro=10, releaselevel='final', serial=0)`

### PR DESCRIPTION
**Pull request by sethtroisi**
fixes #12696

x matches before and after this change.
```
from scipy import optimize
import numpy as np
optimize.fminbound(lambda x: np.array([x**2]), -np.pi, np.pi, disp=False)
```

### CODE REVIEW

File: scipy/optimize/optimize.py [hunk 1]
```diff
@@ -1973,7 +1973,7 @@ def _minimize_scalar_bounded(func, bounds, args=(),
             step = '       golden'
 
         si = np.sign(rat) + (rat == 0)
-        x = xf + si * np.max([np.abs(rat), tol1])
+        x = xf + si * np.maximum(np.abs(rat), tol1)
```
Comment (sethtroisi) [inline scipy/optimize/optimize.py:1976]: `rat` can be a scalar or nd.array(scalar) or nd.array([scalar]).

**Review by andyfaff — COMMENTED** [review]
Can you add a test?

**Comment by sethtroisi** [comment]
> Can you add a test?

What would you like it to test?

The regression test (IMO) would be that this doesn't cause a warning in numpy >= 1.19 and I didn't see a lot of value in that test but I am happy to add if it is commonly done.

**Comment by andyfaff** [comment]
The OP had an example that displayed the issue.

**Comment by sethtroisi** [comment]
I added a test and verified it fails before my change (and passes afterwards).

**Review by peterbell10 — APPROVED** [review]

**Comment by peterbell10** [comment]
Thanks @sethtroisi.

**Pull request merged**